sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -83,33 +83,39 @@ abstract class StringRegexExpression extends BinaryExpression
           % matches zero or more characters in the input (similar to .* in posix regular
           expressions)
 
-          The escape character is '\'. If an escape character precedes a special symbol or another
-          escape character, the following character is matched literally. It is invalid to escape
-          any other character.
-
           Since Spark 2.0, string literals are unescaped in our SQL parser. For example, in order
           to match "\abc", the pattern should be "\\abc".
 
           When SQL config 'spark.sql.parser.escapedStringLiterals' is enabled, it fallbacks
           to Spark 1.6 behavior regarding string literal parsing. For example, if the config is
           enabled, the pattern to match "\abc" should be "\abc".
+      * escape - a optional string. The default escape character is the '\' . If an escape character
+          precedes a special symbol or another escape character, the following character is matched
+          literally. It is invalid to escape any other character. You can specify '' as escape
+          character so that disables the escape mechanism, which makes it impossible to turn off
+          the special meaning of underscore and percent signs in the pattern.
```

Comment:
I like the previous one.

Reply:
@maropu I'm very glad see your reply. You means this comment should reverse to previous? I adds some description of `''` specified as escape character, we don't need it?

Reply:
> `You can specify '' as escape character so that disables the escape mechanism...`
Is this behaviour the same with the other DBMSs?

Reply:
The behavior references the `PostgreSQL` only. 
`MySQL`, `Oracle`, `Vertica` and `PostgreSQL` haven't unify behavior.

Reply:
Could you show us these behivours?

Reply:
The docs https://www.postgresql.org/docs/11/functions-matching.html exists the description contains the content `It's also possible to select no escape character by writing ESCAPE ''. This effectively disables the escape mechanism`.
```
SELECT 'a_%b' LIKE 'a\\__b' ESCAPE '', 'addb' LIKE 'a_%b' ESCAPE '', 'addb' LIKE 'a\\__b' ESCAPE '', 'addb' LIKE 'a%\\%b' ESCAPE '', 
       'a_%b' LIKE 'a%\\%b' ESCAPE '', 'addb' LIKE 'a%' ESCAPE '', 'addb' LIKE '**' ESCAPE '', 'abc' LIKE 'a%' ESCAPE '',
       'abc' LIKE 'b%' ESCAPE '', 'abc' LIKE 'bc%' ESCAPE '', 'a\nb' LIKE 'a_b' ESCAPE '', 'ab' LIKE 'a%b' ESCAPE '',
       'a\nb' LIKE 'a%b' ESCAPE '';

// result is:
f	t	f	f	f	t	f	t	f	f	f	t	t
```

Reply:
Before that, I just want to make sure that that behaivour is consistent among the other DBMSs.

Reply:
```
SELECT 'a_%b' LIKE 'a\__b', 'a_%b' LIKE 'a\__b' ESCAPE '', 'addb' LIKE 'a_%b', 'addb' LIKE 'a_%b' ESCAPE '';

//PostGresql result is:
t	f	t	t
// MySQL result is:
1     1       1      1
```

Reply:
oh... the standard says something about this?

Reply:
ANSI description is:
```
<character like predicate> ::=
<row value predicand> <character like predicate part 2>
<character like predicate part 2> ::=
[ NOT ] LIKE <character pattern> [ ESCAPE <escape character> ]
<character pattern> ::=
<character value expression>
<escape character> ::=
<character value expression>
```
There are also a lot note: ```
If an <escape character> is specified, then:
1) If the length in characters of ECV is not equal to 1, then an exception condition is raised:
data exception — invalid escape character.
```

Reply:
If we follow `1)`, we should throw an exception for the empty case?

Reply:
> If we follow `1)`, we should throw an exception for the empty case?

Yes. Should we follow which behavior?

Reply:
I like the behivour, but we should wait for the other review's comment. cc: @dongjoon-hyun @gatorsmile

Reply:
@dongjoon-hyun @gatorsmile Could you take a look at this discussion @maropu made a suggestion.

Reply:
@maropu Shall we continue to wait other review's comment?

Reply:
For now, can you fix code to throw an exception for the case?

Reply:
> For now, can you fix code to throw an exception for the case?

OK. we make a decision follow the ANSI behavior.